scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -384,56 +385,72 @@ static s32 pick_idle_cpu_from(const struct cpumask *cand_cpumask, s32 prev_cpu,
 	return scx_bpf_pick_idle_cpu(cand_cpumask, 0);
 }
 
-s32 BPF_STRUCT_OPS(layered_select_cpu, struct task_struct *p, s32 prev_cpu, u64 wake_flags)
+static __always_inline
+bool should_try_preempt_first(s32 cand, struct layer *layer,
+			      const struct cpumask *layered_cpumask)
+{
+	struct cpu_ctx *cand_cctx, *sib_cctx;
+	s32 sib;
+
+	if (!layer->preempt || !layer->preempt_first)
```

Comment:
Maybe I'm missing something here, but shouldn't be `&&` instead of `||`?

Reply:
Hmmm... so, it's saying "if the layer is not preempting or preempt_first is not enabled, tell the caller to not try to preempt first". ie. Both need to be enabled to try preempt_first, no?

Reply:
@htejun maybe to simplify we could force `preempt_first` to be `false` if the layer isn't specified with `"preempt": true`? Or force `preempt: true` to be true if `preempt_first` is set.

Reply:
> Hmmm... so, it's saying "if the layer is not preempting or preempt_first is not enabled, tell the caller to not try to preempt first". ie. Both need to be enabled to try preempt_first, no?

Oh I see, now it's clear. Not a blocker obviously, but I like @Byte-Lab suggestion,  I think it makes the code easier to understand for those who aren't familiar with the whole logic. Thanks!